gui/gui/src/ts/components/pages/NodeSearchResultsPage.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from "react";
+import {Grid, Typography} from "@material-ui/core";
+import {Frame} from "components/frame/Frame";
+
+import {useQuery} from "@apollo/react-hooks";
+import {
+  NodeSearchResultsPageQuery,
+  NodeSearchResultsPageQueryVariables,
+} from "api/queries/types/NodeSearchResultsPageQuery";
+import * as NodeSearchResultsPageQueryDocument from "api/queries/NodeSearchResultsPageQuery.graphql";
+import {NodeTable} from "components/data/NodeTable";
+
+import * as ReactLoader from "react-loader";
+import {useLocation, useHistory} from "react-router-dom";
+import * as qs from "qs";
+
+interface NodeSearchVariables {
+  text: string;
+  offset?: number;
+  limit?: number;
+}
+
+class QueryStringNodeSearchVariables implements NodeSearchVariables {
+  private constructor(
+    public readonly text: string,
+    public readonly offset: number = 0,
+    public readonly limit: number = 10
+  ) {}
+
+  get page() {
+    return this.offset / this.limit;
+  }
+
+  get object() {
+    return {text: this.text, offset: this.offset, limit: this.limit};
+  }
+
+  static parse(queryString: string) {
+    const {text, offset, limit} = (qs.parse(queryString, {
+      ignoreQueryPrefix: true,
+    }) as unknown) as {
+      text: string;
+      offset: string;
+      limit: string;
+    };
+    return new QueryStringNodeSearchVariables(
+      text,
+      offset === undefined ? undefined : +offset,
+      limit === undefined ? undefined : +limit
+    );
+  }
+
+  stringify() {
+    return qs.stringify(this.object, {addQueryPrefix: true});
+  }
+
+  replace({text, offset, limit}: Partial<NodeSearchVariables>) {
+    return new QueryStringNodeSearchVariables(
+      text !== undefined ? text : this.text,
+      offset !== undefined ? offset : this.offset,
+      limit !== undefined ? limit : this.limit
+    );
+  }
+}
+
+export const NodeSearchResultsPage: React.FunctionComponent<{}> = ({}) => {
+  const history = useHistory();
+
+  const location = useLocation();
+
+  const searchVariables = QueryStringNodeSearchVariables.parse(location.search);
+
+  const [count, setCount] = React.useState<number | null>(null);
+
+  const {data, loading} = useQuery<
+    NodeSearchResultsPageQuery,
+    NodeSearchResultsPageQueryVariables
+  >(NodeSearchResultsPageQueryDocument, {
+    variables: {...searchVariables.object, withCount: count === null},
+  });
+
+  if (data?.matchingNodesCount) {
```

Comment:
It's fine to use ReactLoader as you did, but you need to check error here, too. See NodePage for how I do that.

Reply:
Ah missed that, added